spec/defines/timer_wrapper_spec.rb

```diff
@@ -0,0 +1,177 @@
+              with_content(%r{User=root})
+            is_expected.to contain_file("/etc/systemd/system/#{title}.timer").
+              with_content(%r{OnCalendar=\*:0/10})
+            is_expected.to contain_Systemd__Unit_file("#{title}.service").
+              that_comes_before("Systemd::Unit_file[#{title}.timer]")
+            is_expected.to contain_Systemd__Unit_file("#{title}.service").
+              that_comes_before("Systemd::Unit_file[#{title}.timer]")
+            is_expected.to contain_Exec("systemd-#{title}.service-systemctl-daemon-reload")
+            is_expected.to contain_Exec("systemd-#{title}.timer-systemctl-daemon-reload")
+            is_expected.to contain_Service("#{title}.timer")
+            is_expected.to contain_Systemd__Daemon_reload("#{title}.service")
+            is_expected.to contain_Systemd__Daemon_reload("#{title}.timer")
+            is_expected.to contain_Systemd__Manage_unit("#{title}.service")
+            is_expected.to contain_Systemd__Manage_unit("#{title}.timer")
+            is_expected.to contain_Systemd__Unit_file("#{title}.timer")
+          end
+        end
+
+        context 'failue when not passing calendar spec' do
+          let :params do
+            {
+              ensure: 'present',
+              command: '/bin/date',
+              user: 'root',
+            }
+          end
+
+          it do
+            is_expected.to compile.and_raise_error(%r{At least one of on_active_sec})
+          end
+        end
+
+        context 'with / in title' do
+          let :title do
+            't/i/t/l/e'
+          end
+          let :params do
+            {
+              ensure: 'present',
+              on_calendar: '*:0/10',
+              command: '/bin/true',
+              user: 'root',
+            }
+          end
+
+          it {
+            is_expected.to compile
+            is_expected.to contain_file('/etc/systemd/system/t_i_t_l_e.timer')
+            is_expected.to contain_systemd__manage_unit('t_i_t_l_e.timer')
+            is_expected.to contain_systemd__unit_file('t_i_t_l_e.timer')
+            is_expected.to contain_file('/etc/systemd/system/t_i_t_l_e.service')
+            is_expected.to contain_systemd__manage_unit('t_i_t_l_e.service')
+            is_expected.to contain_systemd__unit_file('t_i_t_l_e.service')
+            is_expected.to contain_service('t_i_t_l_e.timer')
+            is_expected.to contain_exec('systemd-t_i_t_l_e.service-systemctl-daemon-reload')
+            is_expected.to contain_exec('systemd-t_i_t_l_e.timer-systemctl-daemon-reload')
+            is_expected.to contain_Systemd__Daemon_reload('t_i_t_l_e.service')
+            is_expected.to contain_Systemd__Daemon_reload('t_i_t_l_e.timer')
+          }
+        end
+
+        context 'ensure absent' do
+          let :params do
+            {
+              ensure: 'absent',
+            }
+          end
+
+          it {
+            is_expected.to contain_Systemd__Manage_unit("#{title}.timer")
```

Comment:
Would be good to verify the ensure is really absent here - same for all the other service and timers.

Reply:
done